Infrastructure/DataImporterService.cs

```diff
@@ -191,32 +194,228 @@ private async Task RemoveGeneralInformationAboutSchools(NtpDbContext dbContext,
+
+            if (existingLaDistrictCoverage == null)
+            {
+                existingTP.LocalAuthorityDistrictCoverage.Add(laDistrictCoverage);
+            }
+            else
+            {
+                existingLaDistrictCoverage = laDistrictCoverage.Adapt(existingLaDistrictCoverage);
+            }
+        }
+    }
+
+    private static void ImportTuitionPartnerSubjectCoverage(NtpDbContext dbContext, TuitionPartner existingTP, TuitionPartner tuitionPartnerToProcess)
+    {
+        var subjectCoveragesToDelete = existingTP.SubjectCoverage.Where(x => !tuitionPartnerToProcess.SubjectCoverage.Any(e => e.TuitionTypeId == x.TuitionTypeId &&
+                                                                                                                            e.SubjectId == x.SubjectId));
+        foreach (var subjectCoverageToDelete in subjectCoveragesToDelete)
+        {
+            dbContext.Entry(subjectCoverageToDelete).State = EntityState.Deleted;
+        }
+
+        foreach (var subjectCoverage in tuitionPartnerToProcess.SubjectCoverage)
+        {
+            var existingSubjectCoverage = existingTP.SubjectCoverage.FirstOrDefault(x => x.TuitionTypeId == subjectCoverage.TuitionTypeId &&
+                                                                                x.SubjectId == subjectCoverage.SubjectId);
+
+            if (existingSubjectCoverage == null)
+            {
+                existingTP.SubjectCoverage.Add(subjectCoverage);
+            }
+            else
+            {
+                existingSubjectCoverage = subjectCoverage.Adapt(existingSubjectCoverage);
+            }
+        }
+    }
+
+    private static void ImportTuitionPartnerPrices(NtpDbContext dbContext, TuitionPartner existingTP, TuitionPartner tuitionPartnerToProcess)
+    {
+        var pricesToDelete = existingTP.Prices.Where(x => !tuitionPartnerToProcess.Prices.Any(e => e.TuitionTypeId == x.TuitionTypeId &&
+                                                                                                e.SubjectId == x.SubjectId &&
+                                                                                                e.GroupSize == x.GroupSize));
+        foreach (var priceToDelete in pricesToDelete)
+        {
+            dbContext.Entry(priceToDelete).State = EntityState.Deleted;
+        }
+
+        foreach (var price in tuitionPartnerToProcess.Prices)
+        {
+            var existingPrice = existingTP.Prices.FirstOrDefault(x => x.TuitionTypeId == price.TuitionTypeId &&
+                                                                    x.SubjectId == price.SubjectId &&
+                                                                    x.GroupSize == price.GroupSize);
+
+            if (existingPrice == null)
+            {
+                existingTP.Prices.Add(price);
+            }
+            else
+            {
+                existingPrice = price.Adapt(existingPrice);
```

Comment:
The value assigned is not used in any execution path. Do we need it?

Reply:
This is used, because it updates the object (by ref) and this is then flagged as modified (if data has changed) with the following check.  It's then subsequently saved. 
                    dbContext.ChangeTracker.Entries().Any(e => e.State == EntityState.Modified ||
                                                                e.State == EntityState.Added ||
                                                                e.State == EntityState.Deleted

I have added a comment to explain this is the case.